src/main/java/uk/gov/companieshouse/web/accounts/service/navigation/NavigatorService.java

```diff
@@ -77,13 +116,51 @@ private Class getNextControllerClass(Class clazz) {
      * @param  clazz the controller class in the chain to begin the scan at
      * @return the previous controller class in the chain
      */
-    private Class getPreviousControllerClass(Class clazz) {
+    private Class getPreviousControllerClass(Class clazz, String ... pathVars) {
         Annotation previousControllerAnnotation = AnnotationUtils.findAnnotation(clazz, PreviousController.class);
         if (previousControllerAnnotation == null) {
             throw new MissingAnnotationException("Missing @PreviousController annotation on " + clazz.toString());
         }
 
-        return ((PreviousController) previousControllerAnnotation).value();
+        Class[] classList = ((PreviousController) previousControllerAnnotation).value();
+
+        if (classList.length > 0) {
```

Comment:
This logic is the same as the above. Could it be moved into a common method?

Reply:
This is a pain due to the casting of some objects that takes place